client.go

```diff
@@ -1371,3 +1386,15 @@ func (c *client) persistInbound(m packets.ControlPacket) {
 func (c *client) pingRespReceived() {
 	atomic.StoreInt32(&c.pingOutstanding, 0)
 }
+
+// checkAndSetFastReconnectCheckStartTime will be called whenever a packet is sent to check the broker's connection status
+func (c *client) checkAndSetFastReconnectCheckStartTime() {
+	fastReconnectTime := c.fastReconnectCheckStartTime.Load().(time.Time)
```

Comment:
How about adding a check if ackTimeout > 0 and wrapping the function body in it?